rust/main/chains/hyperlane-sealevel/src/mailbox.rs

```diff
@@ -508,133 +463,40 @@ impl Mailbox for SealevelMailbox {
-            });
-        let ixn_data = ixn
-            .into_instruction_data()
-            .map_err(ChainCommunicationError::from_other)?;
-
-        // Craft the accounts for the transaction.
-        let mut accounts: Vec<AccountMeta> = vec![
-            AccountMeta::new_readonly(payer.pubkey(), true),
-            AccountMeta::new_readonly(Pubkey::from_str(SYSTEM_PROGRAM).unwrap(), false),
-            AccountMeta::new(self.inbox.0, false),
-            AccountMeta::new_readonly(process_authority_key, false),
-            AccountMeta::new(processed_message_account_key, false),
-        ];
-        accounts.extend(ism_getter_account_metas);
-        accounts.extend([
-            AccountMeta::new_readonly(Pubkey::from_str(SPL_NOOP).unwrap(), false),
-            AccountMeta::new_readonly(ism, false),
-        ]);
-
-        // Get the account metas required for the ISM.Verify instruction.
-        let ism_verify_account_metas = self
-            .get_ism_verify_account_metas(ism, metadata.into(), encoded_message)
+        let tx = self
+            .provider
+            .rpc()
+            .build_estimated_tx_for_instruction(
+                process_instruction,
+                self.get_payer()?,
+                &*self.tx_submitter,
+                &*self.priority_fee_oracle,
+            )
             .await?;
-        accounts.extend(ism_verify_account_metas);
 
-        // The recipient.
-        accounts.extend([AccountMeta::new_readonly(recipient, false)]);
+        tracing::info!(?tx, "Created sealevel transaction to process message");
 
-        // Get account metas required for the Handle instruction
-        let handle_account_metas = self.get_handle_account_metas(message).await?;
-        accounts.extend(handle_account_metas);
+        let signature = self.tx_submitter.send_transaction(&tx, true).await?;
 
-        let inbox_instruction = Instruction {
-            program_id: self.program_id,
-            data: ixn_data,
-            accounts,
-        };
-        instructions.push(inbox_instruction);
-        let recent_blockhash = self
-            .rpc()
-            .get_latest_blockhash_with_commitment(commitment)
-            .await?;
+        tracing::info!(?tx, ?signature, "Sealevel transaction sent");
 
-        let txn = Transaction::new_signed_with_payer(
-            &instructions,
-            Some(&payer.pubkey()),
-            &[payer],
-            recent_blockhash,
-        );
+        let send_instant = std::time::Instant::now();
 
-        tracing::info!(?txn, "Created sealevel transaction to process message");
+        // Wait for the transaction to be confirmed.
+        self.rpc().wait_for_transaction_confirmation(&tx).await?;
 
-        let signature = self.send_and_confirm_transaction(&txn).await?;
-
-        tracing::info!(?txn, ?signature, "Sealevel transaction sent");
+        tracing::info!(?tx, ?signature, time_to_confirm=?send_instant.elapsed(), "Sealevel transaction confirmed");
```

Comment:
isn't the `send_instant.elapsed()` more of a result of the commitment level used? Especially since there's also a confirm check below, suggesting we don't yet have confirmation here

Reply:
hmm yes, I'm adding a comment about this but gonna leave logic as is atm. I think it's probably still useful